corehq/ex-submodules/casexml/apps/phone/data_providers/case/livequery.py

```diff
@@ -262,6 +262,16 @@ def update_open_and_deleted_ids(related):
 
         with timing_context("compile_response(%s cases)" % len(sync_ids)):
             iaccessor = PrefetchIndexCaseAccessor(accessor, indices)
+            metrics_histogram(
```

Comment:
any reason you moved this? I moderately preferred the other place because it is right next to the other case load metrics calculated during livequery, rather than splitting them across two different functions (at two different levels of the call stack as you mention)

Reply:
I needed access to `restore_state.last_sync_log` so my choices were to pass that down, or pull this up. I chose to pull this up because of the two it made more sense to me. This metric says "A livequery restore happened and it had this many cases in it", and so it makes sense to me to have this be in the main body rather than in a helper function. Passing another value down into the helper function so it could add it to the metric seemed to be more of a contortion than this

Reply:
Yeah makes sense. missed that you would need to pass that down. that seems fair. How do you feel about pulling the other one up?

Reply:
the other one does something a bit different and is meant to track _work as it happens_ rather than higher level data about a request. (If pulling in all the docs takes 2 minutes, that one would emit pretty evenly over those two minutes, whereas this one would emit all at once right at the beginning of those two minutes.) You'll notice that it defines `track_load` and then calls `track_load` within the while loop to achieve that, so it's pretty clearly the intention, and I didn't want to mess with that